lang/java/avro/src/main/java/org/apache/avro/file/DataFileStream.java

```diff
@@ -351,16 +349,27 @@ ByteBuffer getAsByteBuffer() {
       return ByteBuffer.wrap(data, offset, blockSize);
     }
 
+    void setBytes(ByteBuffer block) {
```

Comment:
this now supports Codecs that return Direct ByteBuffers.  Earlier versions of the ZstandardCodec were using APIs that returned direct buffers, which exploded here.